bin/diagnostic.js

```diff
@@ -1 +1,15 @@
 'use strict';
+
+
+//We'll learn about require later in the course
+const ask = require('../lib/ask.js');
+
+let answer = 10;
+
+while (answer < 10) {
+  answer = ask("Guess my secret? ");
+  if (answer !== 1 && answer <= 10) {
+   console.log("Your guess needs to be between 1 and 10.");
+   break;
+  }
+}
```

Comment:
Great effort on this, you were very close. Here's a possible solution:

while (answer !== '') {
  answer = ask("Guess my secret? ");
  if (answer >=1 && answer <=10) {
   console.log("You guessed it!");
   break;
 }else{
   console.log('Please try a number between 1 and 10')
 }
}